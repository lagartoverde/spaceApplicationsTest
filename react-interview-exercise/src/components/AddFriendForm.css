:local(.AddFriendForm) {
  border-radius: 0;
  border-color: #ABAAAA;
  border-left: 0;
  border-right: 0;
}

:local(.addFriendForm) {
  display: flex;
  flex-direction: column;
}

:local(.genderSelection) {
  margin-top: 10px;
  margin-bottom: 10px;
  color: white;
  display: flex;
  justify-content: center;
}

:local(.radioButton) {
  cursor: pointer;
}