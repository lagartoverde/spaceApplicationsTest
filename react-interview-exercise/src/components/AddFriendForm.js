import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import styles from './AddFriendForm.css';

class AddFriendForm extends Component {
  constructor (props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
      gender: this.props.gender || '',
    };
    this.handleNameChange = this.handleNameChange.bind(this)
    this.handleGenderChange = this.handleGenderChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  render () {
    return (
      <form onSubmit={this.handleSubmit} className={styles.addFriendForm}>
        <input
          type="text"
          autoFocus="true"
          id='name'
          className={classnames('form-control', styles.AddFriendForm)}
          placeholder="Type the name of a friend to be added"
          value={this.state.name}
          onChange={this.handleNameChange}
        />
        <div className={styles.genderSelection}>
          <label className="radio-inline">
            <input 
              type="radio" 
              value="male"
              id='maleRadioButton'
              name='gender'
              checked={this.state.gender === 'male'} 
              onChange={this.handleGenderChange}
              className={styles.radioButton}
            />
            Male
          </label>
          <label className="radio-inline">
            <input 
              type="radio" 
              value="female"
              id='femaleRadioButton'
              name='gender' 
              checked={this.state.gender === 'female'}
              onChange={this.handleGenderChange}
              className={styles.radioButton}
            />
            Female
          </label>
          <label className="radio-inline">
            <input 
              type="radio" 
              value="other" 
              id='otherRadioButton'
              name='gender'
              checked={this.state.gender === 'other'}
              onChange={this.handleGenderChange}
              className={styles.radioButton}
            />
            Other
          </label>
        </div>
        <button className='btn btn-primary'>Submit</button>
      </form>
    );
  }

  handleNameChange (e) {
    this.setState({ name: e.target.value });
  }

  handleGenderChange(e) {
    this.setState({gender: e.target.value})
  }

  handleSubmit (e) {
    e.preventDefault()
    const {name, gender} = this.state
    this.props.addFriend({name, gender});
    this.setState({ name: '', gender: undefined });
  }

}

AddFriendForm.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriendForm
