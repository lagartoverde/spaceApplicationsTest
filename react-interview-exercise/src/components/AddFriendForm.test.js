import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import Enzyme, { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-15';
import AddFriendForm from './AddFriendForm';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddFriendForm  addFriend={() => {}}/>, div);
});

it('calls the add friend function', () => {
  const addFriendFn = jest.fn();
  const wrapper = mount(<AddFriendForm addFriend={addFriendFn}/>);
  const form = wrapper.find('form');
  form.simulate('submit')
  expect(addFriendFn).toHaveBeenCalledTimes(1);
});

it('sends the correct values to the add friend function', () => {
  const addFriendFn = jest.fn(data => data);
  const wrapper = mount(<AddFriendForm addFriend={addFriendFn}/>);
  const nameInput = wrapper.find('#name')
  const maleRadioButton = wrapper.find('#maleRadioButton')
  const form = wrapper.find('form');
  nameInput.simulate('change', {target: {value: 'Oscar'}});
  maleRadioButton.simulate('change', {target: {value: 'male'}});
  form.simulate('submit');
  expect(addFriendFn).toHaveBeenCalledTimes(1);
  expect(addFriendFn.mock.calls[0][0]).toEqual({name: 'Oscar', gender: 'male'});
})

it('component matches the snapshot', () => {
  const component = renderer
    .create(<AddFriendForm  addFriend={() => {}}/>)
    .toJSON();
  expect(component).toMatchSnapshot();
})