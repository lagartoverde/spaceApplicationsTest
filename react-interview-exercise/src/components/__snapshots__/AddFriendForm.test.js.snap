exports[`test component matches the snapshot 1`] = `
<form
  className={undefined}
  onSubmit={[Function]}>
  <input
    autoFocus="true"
    className="form-control"
    id="name"
    onChange={[Function]}
    placeholder="Type the name of a friend to be added"
    type="text"
    value="" />
  <div
    className={undefined}>
    <label
      className="radio-inline">
      <input
        checked={false}
        className={undefined}
        id="maleRadioButton"
        name="gender"
        onChange={[Function]}
        type="radio"
        value="male" />
      Male
    </label>
    <label
      className="radio-inline">
      <input
        checked={false}
        className={undefined}
        id="femaleRadioButton"
        name="gender"
        onChange={[Function]}
        type="radio"
        value="female" />
      Female
    </label>
    <label
      className="radio-inline">
      <input
        checked={false}
        className={undefined}
        id="otherRadioButton"
        name="gender"
        onChange={[Function]}
        type="radio"
        value="other" />
      Other
    </label>
  </div>
  <button
    className="btn btn-primary">
    Submit
  </button>
</form>
`;
